PolyEngine/Engine/Src/World.cpp

```diff
@@ -49,7 +49,7 @@ void World::DestroyEntity(const UniqueID& entityId)
 	TransformComponent* transform = ent->GetComponent<TransformComponent>();
 	if (transform)
 	{
-		for (auto it : transform->GetChildren()) DestroyEntity(it->GetOwnerID());
```

Comment:
You can and should call `DestroyEntity` method here it's private, but you are calling it from within the class.